stream-chat-android-client/src/main/java/io/getstream/chat/android/client/setup/InitializationCoordinator.kt

```diff
@@ -0,0 +1,42 @@
+package io.getstream.chat.android.client.setup
+
+import io.getstream.chat.android.client.models.User
+
+// Make it internal once ChatDomain is merged with ChatClient.
+public class InitializationCoordinator private constructor() {
+
+    private val userDisconnectedListeners: MutableList<(User?) -> Unit> = mutableListOf()
+    private val userConnectedListeners: MutableList<(User) -> Unit> = mutableListOf()
+    private val databasedCreatedListeners: MutableList<() -> Unit> = mutableListOf()
```

Comment:
As discussed on Slack - let's add TODO

Reply:
In the end, there's no need for `databasedCreatedListeners`. The initialization moved to `StreamOfflinePluginFactory`